packages/api/src/webhooks/cannon.ts

```diff
@@ -514,23 +515,19 @@ export default class WebhookCannon {
       // there was an update after the delayed event was sent, so sleep a few
       // secs (up to USER_SESSION_TIMEOUT) and re-check if it actually stopped.
       await sleep(5000 + (lastSeen - activeThreshold));
-      return this.handleRecordingReadyChecks(sessionId, mp4Url, true);
+      return this.handleRecordingReadyChecks(sessionId, true);
     }
 
-    const res = await this.db.stream.update(
-      [sql`id = ${sessionId}`, sql`data->>'isActive' != 'false'`],
-      { isActive: false },
-      { throwIfEmpty: false }
-    );
-    if (res.rowCount < 1) {
+    const res = await this.db.asset.get(sessionId);
+    if (res) {
```

Comment:
Can you make sure to gracefully handle duplicate primary key errors from the `createAsset` function? Otherwise we may still be subject to race conditions here